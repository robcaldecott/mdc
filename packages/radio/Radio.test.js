import React from "react";
import { Radio } from ".";
import renderer from "react-test-renderer";

it("renders", () => {
  expect(renderer.create(<Radio />)).toMatchSnapshot();
});

it("renders disabled", () => {
  expect(renderer.create(<Radio disabled />)).toMatchSnapshot();
});

it("renders with a label", () => {
  expect(renderer.create(<Radio id="radio">Label</Radio>)).toMatchSnapshot();
});

it("renders alignEnd", () => {
  expect(
    renderer.create(
      <Radio id="radio" alignEnd>
        Label
      </Radio>
    )
  ).toMatchSnapshot();
});
