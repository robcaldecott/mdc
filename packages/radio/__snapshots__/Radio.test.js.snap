// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<div
  className="mdc-radio"
>
  <input
    className="mdc-radio__native-control"
    disabled={undefined}
    id={undefined}
    type="radio"
  />
  <div
    className="mdc-radio__background"
  >
    <div
      className="mdc-radio__outer-circle"
    />
    <div
      className="mdc-radio__inner-circle"
    />
  </div>
</div>
`;

exports[`renders alignEnd 1`] = `
<div
  className="mdc-form-field mdc-form-field--align-end"
>
  <div
    className="mdc-radio"
  >
    <input
      className="mdc-radio__native-control"
      disabled={undefined}
      id="radio"
      type="radio"
    />
    <div
      className="mdc-radio__background"
    >
      <div
        className="mdc-radio__outer-circle"
      />
      <div
        className="mdc-radio__inner-circle"
      />
    </div>
  </div>
  <label
    htmlFor="radio"
  >
    Label
  </label>
</div>
`;

exports[`renders disabled 1`] = `
<div
  className="mdc-radio mdc-radio--disabled"
>
  <input
    className="mdc-radio__native-control"
    disabled={true}
    id={undefined}
    type="radio"
  />
  <div
    className="mdc-radio__background"
  >
    <div
      className="mdc-radio__outer-circle"
    />
    <div
      className="mdc-radio__inner-circle"
    />
  </div>
</div>
`;

exports[`renders with a label 1`] = `
<div
  className="mdc-form-field"
>
  <div
    className="mdc-radio"
  >
    <input
      className="mdc-radio__native-control"
      disabled={undefined}
      id="radio"
      type="radio"
    />
    <div
      className="mdc-radio__background"
    >
      <div
        className="mdc-radio__outer-circle"
      />
      <div
        className="mdc-radio__inner-circle"
      />
    </div>
  </div>
  <label
    htmlFor="radio"
  >
    Label
  </label>
</div>
`;
